### PR TITLE
[RDY] Applying some of the LanguageTool suggestions

### PR DESCRIPTION
**Pull request by int-ua**
#### Summary

SUMMARY: Bugfixes "Proofreading with LanguageTool, second batch"

#### Purpose of change

Fix errors and apply style improvement suggestions.

#### Describe the solution

(simplified)
python3 table.py -f csv description > languagetool_input.csv

Some of the changes applied with mass `grep` and `sed`, beware. Edit: but most of them were reverted now. 

Stopped at "meathook" line around the middle for now.

Please squash on merging.

#### Describe alternatives you've considered

Not feeding my urge to fix them? No.

#### Testing

~~my patience~~

#### Additional context

#42578

### CODE REVIEW

**Comment by int-ua** [comment]
> int-ua requested a review from I-am-Erk as a code owner 2 minutes ago
I did not. Guess that's something from Actions.

File: data/json/items/book/cooking.json [hunk 1]
```diff
@@ -254,7 +254,7 @@
     "id": "cookbook_daintydishes",
     "type": "BOOK",
     "name": { "str": "Dainty Dishes Fit for a King", "str_pl": "copies of Dainty Dishes" },
-    "description": "This cookbook contains many fanciful recipes.  Some of them are even worth the trouble to actually prepare.  Lovely pen and ink illustrations of overweight knights and gluttonous monks line the pages.",
+    "description": "This cookbook contains many fanciful recipes.  Some m are even worth the trouble to actually prepare.  Lovely pen and ink illustrations of overweight knights and gluttonous monks line the pages.",
```
Comment (Night-Pryanik) [inline data/json/items/book/cooking.json:257]: ?

File: data/json/items/tool/science.json [hunk 2]
```diff
@@ -463,7 +463,7 @@
     "id": "filter_paper_qualitative",
     "type": "TOOL",
     "name": { "str_sp": "qualitative filter paper" },
-    "description": "filter paper for seperating solids from liquids.  This filter paper has a rating of 2 microns, seperating more solids out than quantitative filter paper but the process takes longer.",
+    "description": "filter paper for separating solids from liquids.  This filter paper has a rating of 2 microns, separating more solids out than quantitative filter paper but the process takes longer.",
```
Comment (Night-Pryanik) [inline data/json/items/tool/science.json:466]: Could you please capitalize the first word in the sentence while you're here?

File: data/json/items/tool/science.json [hunk 3]
```diff
@@ -479,7 +479,7 @@
     "id": "filter_paper_quantitative",
     "type": "TOOL",
     "name": { "str_sp": "quantitative filter paper" },
-    "description": "filter paper for seperating solids from liquids.  This filter paper has a rating of 20 microns, seperating less solids out than qualitative filter paper but the process is much faster.",
+    "description": "filter paper for separating solids from liquids.  This filter paper has a rating of 20 microns, separating less solids out than qualitative filter paper but the process is much faster.",
```
Comment (Night-Pryanik) [inline data/json/items/tool/science.json:482]: Here too.

File: data/json/npcs/TALK_COMMON_ALLY.json [hunk 4]
```diff
@@ -840,7 +840,7 @@
         "effect": "revert_activity"
       },
       {
-        "text": "Please sort out some of these items nearby.",
+        "text": "Please sort out some se items nearby.",
```
Comment (int-ua) [inline data/json/npcs/TALK_COMMON_ALLY.json:843]: :/

File: data/json/items/tool/science.json (hunk 2 above)
Comment (int-ua) [reply data/json/items/tool/science.json:466]: Yeap

File: data/json/items/book/cooking.json (hunk 1 above)
Comment (int-ua) [reply data/json/items/book/cooking.json:257]: !

File: data/json/npcs/TALK_COMMON_ALLY.json [hunk 5]
```diff
@@ -840,7 +840,7 @@
         "effect": "revert_activity"
       },
       {
-        "text": "Please sort out some of these items nearby.",
+        "text": "Please sort out some these items nearby.",
```
Comment (int-ua) [inline data/json/npcs/TALK_COMMON_ALLY.json:843]: !

File: data/json/npcs/hints.json [hunk 6]
```diff
@@ -64,7 +64,7 @@
       "If you get a pair of night vision goggles, hold on to them!  A flashlight will give you away, but with goggles you can be sneaky.  Beware that some types of zombies are camouflaged against it, and require proper light to be visible.",
       "I know they look dumb, but wearing a fanny pack gives you that extra bit of storage without encumbering you.",
       "Backpacks let you carry lots of stuff, but they limit your movement a lot.  If you have to fight a zombie at close range, don't wear one, or at least drop it on the ground before the fight.",
-      "Don't underestimate a good book.  Not only will it keep you company on lonely nights, but you can learn a hell of a lot from some of them.",
+      "Don't underestimate a good book.  Not only will it keep you company on lonely nights, but you can learn a hell of a lot from some m.",
```
Comment (int-ua) [inline data/json/npcs/hints.json:67]: :(

File: data/json/npcs/isherwood_farm/NPC_Claire_Isherwood.json [hunk 7]
```diff
@@ -199,7 +199,7 @@
   {
     "type": "talk_topic",
     "id": "TALK_BARRY",
-    "dynamic_line": "Some of their cows were killed before we realized what was happening, Barry and Carlos went into the forest to find the predator but it was like nothing they'd ever seen.  Carlos may be willing to tell you more.",
+    "dynamic_line": "Some ir cows were killed before we realized what was happening, Barry and Carlos went into the forest to find the predator but it was like nothing they'd ever seen.  Carlos may be willing to tell you more.",
```
Comment (int-ua) [inline data/json/npcs/isherwood_farm/NPC_Claire_Isherwood.json:202]: !

File: data/json/npcs/missiondef.json [hunk 8]
```diff
@@ -564,7 +564,7 @@
       "inquire": "Any leads on where a camp might be?",
       "success": "I guess this wasn't as bright an idea as I thought.",
       "success_lie": "Thanks for trying…  I guess.",
-      "failure": "I bet some of them are still out there…"
+      "failure": "I bet some m are still out there…"
```
Comment (int-ua) [inline data/json/npcs/missiondef.json:567]: ;(

File: data/json/npcs/refugee_center/surface_refugees/NPC_Jenny_Forcette.json [hunk 9]
```diff
@@ -210,7 +210,7 @@
...
+    "dynamic_line": "They're pretty big.  This isn't something you'd carry around with you.  They fire .30 cal bullets using compressed air that we'd make inside the basement and then pipe up to individual storage tanks for the guns that would be good for a few hundred shots each.  The design should be capable of auto or semi-auto fire, at a range that's pretty comparable to a gunpowder-based weapon.  It takes out some most limited parts of ranged weapons: no gunpowder, no brass casings, just lead that we melt into my mold.  It's not soundless, but it's not as loud as a chemical weapon.  There are tons of advantages.  Only trouble is, I can't convince the Free Merchants to give me the parts I need.",
```
Comment (int-ua) [inline data/json/npcs/refugee_center/surface_refugees/NPC_Jenny_Forcette.json:213]: ...

File: data/json/npcs/refugee_center/surface_refugees/NPC_Pablo_Nunez.json [hunk 10]
```diff
@@ -166,7 +166,7 @@
...
+    "dynamic_line": "From the bus?  A few of us made it in the end; the Singh family were on that bus with us.  We were some last refugees to arrive.  They put us into a room for processing, where a bunch of the wounded were being handled as best as they could manage.  They were short staffed though, and they missed one.  We all thought he was just having a nap, but he wasn't, and when he got back up as a zombie, there was a huge panic.  Gunshots in a small, crowded room, and trampling… it was really ugly.  We were close to the door and managed to get out right away, but then a couple panicking people closed the place off and left innocent people in there for what seemed like ages, calling it an 'outbreak'.  By the time leveler heads prevailed, there weren't many people left in there to recover.",
```
Comment (int-ua) [inline data/json/npcs/refugee_center/surface_refugees/NPC_Pablo_Nunez.json:169]: in progress

File: data/json/furniture_and_terrain/terrain-doors.json [hunk 11]
```diff
@@ -1803,7 +1803,7 @@
     "type": "terrain",
     "id": "t_door_boarded_damaged",
     "name": "boarded up damaged door",
-    "description": "A boarded up wooden door, except some of the planks are fractured or coming loose.  Without repairs, it could be broken down easily.",
+    "description": "A boarded up wooden door, except some planks are fractured or coming loose.  Without repairs, it could be broken down easily.",
```
Comment (int-ua) [inline data/json/furniture_and_terrain/terrain-doors.json:1806]: `except for some planks that are`?

**Comment by int-ua** [comment]
Finished looking through the changes, now we need opinions [of native speakers].

File: data/json/furniture_and_terrain/terrain-doors.json (hunk 11 above)
Comment (Qrox) [reply data/json/furniture_and_terrain/terrain-doors.json:1806]: `except that some planks are`, probably.

File: data/json/items/gunmod/mount.json [hunk 12]
```diff
@@ -3,7 +3,7 @@
     "id": "grip_mount",
     "type": "GUNMOD",
     "name": { "str": "replaceable furniture kit" },
-    "description": "This is a kit consisting of various steel and plastic parts; when installed, it would permanently modify and partially replace some of the weapon's furniture so that it could be easily changed if needed.  This allows installing any kind of more advanced grips or other furniture.",
+    "description": "This is a kit consisting of various steel and plastic parts; when installed, it would permanently modify and partially replace some weapon's furniture so that it could be easily changed if needed.  This allows installing any kind of more advanced grips or other furniture.",
```
Comment (andrei8l) [inline data/json/items/gunmod/mount.json:6]: Here, the sentence refers to "some of the furniture", not "some of the weapon". The edit changes the meaning

**Comment by andrei8l** [comment]
I'm not a native speaker and a lot of the sentences seem worse without "some of the". I checked in languagetool and it says
```
if the text is a generality, 'of the' is not necessary'
```
but in many cases here the texts were referring to specific things.

File: data/json/items/gunmod/mount.json (hunk 12 above)
Comment (int-ua) [reply data/json/items/gunmod/mount.json]: Reverted, thank you

**Comment by int-ua** [comment]
Yes, I shouldn't have applied that by a script. Anyway, please comment where the changes make the text worse.

File: data/json/mutations/mutations.json [hunk 13]
```diff
@@ -4495,7 +4495,7 @@
     "points": -4,
     "visibility": 8,
     "mixed_effect": true,
-    "description": "You're only half as tall as you used to be!  The weight of things you once found easy to carry is now unbearable, clothes are now twice as encumbering for you unless you refit them (since you're half their size), and your hit points are heavily reduced.  However, your movement is silent, and your dodge skill is a little higher.",
+    "description": "You are only half as tall as you used to be!  The weight of things you once found easy to carry is now unbearable, clothes are now twice as encumbering for you unless you refit them (since you're half their size), and your hit points are heavily reduced.  However, your movement is silent, and your dodge skill is a little higher.",
```
Comment (Night-Pryanik) [inline data/json/mutations/mutations.json:4498]: That's not an error?

Comment (int-ua) [reply data/json/mutations/mutations.json:4498]: AFAIK, technically, `You're` is correct too, it was highlighted for a different reason (false positive) and I've decided to tune it. Should I revert it?

Comment (Night-Pryanik) [reply data/json/mutations/mutations.json:4498]: Yes, please. The change is minor, but it'll break the existing translation with no profit.

**Comment by I-am-Erk** [comment]
I don't agree with a lot of the removals of `of the`. They frequently change nuance.

**Comment by int-ua** [comment]
Ok, should I revert all of them?

**Comment by int-ua** [comment]
Created this question about "some of the" of StackExchange https://ell.stackexchange.com/q/256402/120123 for further discussion, went through changes again and reverted more.

**Comment by int-ua** [comment]
Ok, after reading https://community.languagetool.org/rule/show/SOME_OF_THE?subId=1&lang=en-US I've reverted most of the "some of the" changes :)

**Comment by jbytheway** [comment]
What sort of false-positive rate are you seeing from this tool?  Would it be feasible to make it a CI check once the things that should be fixed have been?  We can probably manually suppress the false-positives if there are few enough.

**Comment by int-ua** [comment]
@jbytheway By default it provides too much suggestions and has a high rate of false positives. But whole classes of rules can be turned off like this: 

```
java -jar languagetool-commandline.jar -l en --json \
-d EN_QUOTES,ENGLISH_WORD_REPEAT_BEGINNING_RULE,CD_NN,DASH_RULE,USE_TO_VERB,THE_WORSE_OF,PUNCTUATION_PARAGRAPH_END,MASS_AGREEMENT,UNIT_SPACE,EN_DIACRITICS_REPLACE,WORD_CONTAINS_UNDERSCORE \
--disablecategories REDUNDANCY,TYPOGRAPHY,STYLE -
```

I'll try to run it again tomorrow and think about automating it more. The biggest problem right now is the need to locate strings in JSON manually.

**Comment by jbytheway** [comment]
Presumably it ought to be possible for the python script extracting the strings to also somehow save metadata about where they were extracted from. (But it might get messy...)

**Comment by kevingranade** [comment]
This pull request has been mentioned on **Cataclysm: Dark Days Ahead**. There might be relevant details there:

https://discourse.cataclysmdda.org/t/mass-proofreading-with-languagetool/21730/6

**Pull request merged**